DemoApp/src/main.cc

```diff
@@ -1,34 +1,117 @@
+        }
+    }
+
+    return true;
+}
+
+std::vector<std::string> split(const std::string &s, char delimiter) {
+    std::vector<std::string> tokens;
+    std::string token;
+    std::istringstream tokenStream(s);
+    while (std::getline(tokenStream, token, delimiter)) {
+        tokens.push_back(token);
+    }
+    return tokens;
+}
+
+std::map<std::string, std::string> optionsMap(const std::vector<std::string> &args) {
+    std::map<std::string, std::string> params;
+
+    for (const auto &arg : args) {
+        if (!startsWith(arg, "--")) {
+            throw std::runtime_error("Invalid argument: " + arg);
+        }
+        auto val = split(arg, '=');
+        if (val.size() != 2) {
+            throw std::runtime_error("Invalid argument: " + arg);
+        }
+
+        params[val[0].substr(2)] = val[1];
+    }
+
+    return params;
+}
+}  // namespace
+
 int main(int argc, char *argv[]) {
     Echo::initEcho(argc, argv);
 
     // clang-format off
     Menu mainMenu{"Main Menu",{
         new Menu{"Send", {
-            new SendText{"Send text"},
-            new SendFile{"Send file"}
+            new SendText{"sendText", "Send text"},
+            new SendFile{"sendFile", "Send file"}
         }},
         new Menu{"Receive", {
-            new ReceiveText{"Receive text"},
-            new ReceiveFile{"Receive file"}
+            new ReceiveText{"recvText", "Receive text"},
+            new ReceiveFile{"recvFile", "Receive file"}
         }},
         new Menu{"Calibrate", {
-            new CalibrateReceive{"Calibrate receiver"},
-            new CalibrateSend{"Send callibration signal"}
+            new CalibrateReceive{"caliRecv", "Calibrate receiver"},
+            new CalibrateSend{"caliSend", "Send callibration signal"}
         }},
         new Menu{"Experimental", {}},
-        new Settings{"Settings"},
-        new Help{"Help"}
+        new Settings{"settings", "Settings"},
+        new Help{"help", "Help"}
     }};
     // clang-format on
 
-    ViewPtr currentView = &mainMenu;
+    auto opt = getOptions(argc, argv);
+    if (opt.empty()) {
+        ViewPtr currentView = &mainMenu;
+
+        while (currentView) {
+            currentView = currentView->execute();
+        }
+        return 0;
+    }
+
+    Utils::setCLI(true);
+
+    if (opt[0] == "--help") {
+        std::cout << "Usage: " << argv[0] << " [action] [--paramName=paramValue]...\n";
+        std::cout << "Available actions and parameters:\n";
+        mainMenu.printHelp();
+        return 0;
+    }
+
+    std::string action = opt[0];
+    opt.erase(opt.begin());
 
-    while (currentView) {
-        currentView = currentView->execute();
+    try {
+        auto args = optionsMap(opt);
+        if (!mainMenu.executeCLI(action, args)) {
+            std::cerr << "Invalid action name!";
+            return 1;
+        }
+        return 0;
+    } catch (const std::runtime_error &e) {
+        std::cerr << "Error: " << e.what();
+        return 1;
     }
 }
```

Comment:
newline :P

Reply:
Not my changes, it was'n there even before this PR. Its your code btw